clang/lib/Driver/ToolChains/SYCL.cpp

```diff
@@ -101,6 +100,18 @@ const char *SYCL::Linker::constructLLVMLinkCommand(Compilation &C,
     StringRef SubArchName, StringRef OutputFilePrefix,
     const InputInfoList &InputFiles) const {
   ArgStringList CmdArgs;
+
+  bool LinkSYCLDeviceLibs = false;
+  for (const auto &II : InputFiles) {
+    StringRef InputFilename =
+        llvm::sys::path::filename(StringRef(II.getFilename()));
+    StringRef InputSuffix = ".o";
+    if (InputFilename.startswith("libsycl-") &&
+        InputFilename.endswith(InputSuffix)) {
+      LinkSYCLDeviceLibs = true;
+      break;
+    }
+  }
```

Comment:
I think it's better to move this code under if statement at line 120 and replace for-loop with standard library function call e.g. `std::find_if`.